fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBottomSheet.kt

```diff
@@ -262,43 +311,120 @@ private fun TranslationsDialogInfoMessage(
     }
 }
 
+@Suppress("MagicNumber")
 @Composable
 private fun TranslationsDropdown(
     header: String,
     modifier: Modifier = Modifier,
+    isInLandscapeMode: Boolean = false,
+    translateLanguages: List<Language>,
 ) {
-    Column(modifier, verticalArrangement = Arrangement.spacedBy(4.dp)) {
+    var expanded by remember { mutableStateOf(false) }
+
+    var selectedLanguage by remember {
+        mutableStateOf(
+            translateLanguages.last().localizedDisplayName,
+        )
+    }
+
+    val density = LocalDensity.current
+
+    var contextMenuWidthDp by remember {
+        mutableStateOf(0.dp)
+    }
+
+    Column(
+        modifier = modifier
+            .clickable {
+                expanded = true
+            }
+            .semantics { role = Role.DropdownList },
+    ) {
         Text(
             text = header,
+            modifier = Modifier.wrapContentSize(),
             color = FirefoxTheme.colors.textPrimary,
             style = FirefoxTheme.typography.caption,
         )
 
+        Spacer(modifier = Modifier.height(4.dp))
+
         Row {
-            Text(
-                text = "English",
-                modifier = Modifier.weight(1f),
-                color = FirefoxTheme.colors.textPrimary,
-                style = FirefoxTheme.typography.subtitle1,
-            )
+            selectedLanguage?.let {
+                Text(
+                    text = it,
+                    modifier = Modifier.weight(1f),
+                    color = FirefoxTheme.colors.textPrimary,
+                    style = FirefoxTheme.typography.subtitle1,
+                )
+            }
 
             Spacer(modifier = Modifier.width(10.dp))
 
-            Icon(
-                painter = painterResource(id = R.drawable.mozac_ic_dropdown_arrow),
-                contentDescription = null,
-                tint = FirefoxTheme.colors.iconPrimary,
-            )
+            Box {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_dropdown_arrow),
+                    contentDescription = null,
+                    modifier = if (expanded) Modifier.rotate(180f) else Modifier.rotate(0f),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+
+                ContextualMenu(
+                    showMenu = expanded,
+                    onDismissRequest = {
+                        expanded = false
+                    },
+                    menuItems = getContextMenuItems(translateLanguages = translateLanguages) {
+                        expanded = false
+                        selectedLanguage = it.localizedDisplayName
+                    },
+                    modifier = Modifier
+                        .onGloballyPositioned { coordinates ->
```

Comment:
Thanks for figuring out these dropdowns! 

I tested with 28 from languages, which scrolled as expected, but there were a few visual anomalies with that amount of languages. For example, it filled the screen height and also overshot the origin of the dropdown. 

* Could you make an adjustment so only at most four choices show up? (Basing number to [show off of this Figma](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?type=design&node-id=7231-206251&mode=design&t=ODMTQIw4jOI1NLMQ-0).) 
* Also, the list of languages started before the originating dropdown box when testing with 28 languages. Could you lock its start origin to the populating box?

Reply:
Done